sdk/servicebus/azure-servicebus/azure/servicebus/common/mixins.py

```diff
@@ -288,6 +289,15 @@ def __init__(self, address, name, shared_access_key_name=None,
             'key_name': shared_access_key_name,
             'shared_access_key': shared_access_key_value}
 
+        transport_type = kwargs.get('transport_type') or None
+        if transport_type is not None:
+            if transport_type.lower() == "amqpoverwebsocket":
+                self.auth_config['transport_type'] = TransportType.AmqpOverWebsocket
```

Comment:
Do we have a compelling reason to accept string transport types?  Given that the actual parameter type of the enum translates to int, it would seem to give conflicting messages to the user; so I wonder if it'd be cleaner to just take it verbatim and populate docstrings with the enum type guidance.

Reply:
Fair point, I only used string because that's how I parsed it out from the connection string. But I will instead convert directly to the enum while parsing the connection string now.

Reply:
Would this suffice?
```python
def test_servicebusclient_from_conn_str_amqpoverwebsocket():
    conn_str = f'Endpoint={test_addr}/;SharedAccessKeyName={test_key_name};SharedAccessKey={test_key};TransportType=AmqpOverWebsocket'
    sb_client = ServiceBusClient.from_connection_string(conn_str)
    assert sb_client.transport_type == TransportType.AmqpOverWebsocket
```